--- ../source/ext/fluentd/lib/fluent/plugin/in_exec.rb	2017-03-02 15:44:49.000000000 -0800
+++ ../source/ext/fluentd/lib/fluent/plugin/in_exec.rb.new	2017-03-02 15:49:09.000000000 -0800
@@ -140,10 +140,11 @@
           io = IO.popen(@command, "r")
           @parser.call(io)
           Process.waitpid(io.pid)
-          sleep @run_interval
         rescue
           log.error "exec failed to run or shutdown child process", error: $!.to_s, error_class: $!.class.to_s
           log.warn_backtrace $!.backtrace
+        ensure
+          sleep @run_interval
         end
       end
     end
