--- ../source/ext/ruby/ext/openssl/extconf.rb	2018-10-19 03:51:43.030635000 -0700
+++ ../source/ext/ruby/ext/openssl/extconf.1.rb	2018-10-19 17:39:08.081097500 -0700
@@ -109,10 +109,17 @@
 Logging::message "=== Checking for OpenSSL features... ===\n"
 # compile options
 
+# check OPENSSL_NO_{SSL2,SSL3_METHOD} macro: on some environment, these symbols
+# exist even if compiled with no-ssl2 or no-ssl3-method.
+unless have_macro("OPENSSL_NO_SSL2", "openssl/opensslconf.h")
+  have_func("SSLv2_method")
+end
+unless have_macro("OPENSSL_NO_SSL3_METHOD", "openssl/opensslconf.h")
+  have_func("SSLv3_method")
+end
+
 # SSLv2 and SSLv3 may be removed in future versions of OpenSSL, and even macros
 # like OPENSSL_NO_SSL2 may not be defined.
-have_func("SSLv2_method")
-have_func("SSLv3_method")
 have_func("TLSv1_1_method")
 have_func("TLSv1_2_method")
 have_func("RAND_egd")
